lib/util/Components.js

```diff
@@ -419,14 +419,75 @@ function componentRule(rule, context) {
       return utils.isReturningJSX(ASTNode, strict) || utils.isReturningNull(ASTNode);
     },
 
+    getComponentNameFromJSXElement(node) {
+      return node.openingElement.name.name;
```

Comment:
what happens in this function if `node` doesn't have an `openingElement`, or if that doesn't have a `name`? (specifically, what about `<>`?) If this requires the type be "JSXElement", please make this function throw explicitly when `node` has the wrong type.

Reply:
`<>` is a JSXFragement rather then JSXElement. I am not able to think of a situation where JSXElement node will not be having openingElement but to be on safe side I have added null checks.